### PR TITLE
Midnight checkout

### PR DESCRIPTION
**Pull request by dfarooq610**
I am unsure if we still need this, but closes #367 (and hopefully this time with no bugs). @ofloody and I paired on this so long ago but we never actually sent in a PR. So this is it. 

My only concern: what do we do to prevent against the queue-clean service automatically checking out an actual office hours session at midnight? Like if there are office hours in some Russian Timezone that crosses the midnight boundary, how do we ensure that we don't screw over their office hours and all the students in it? 

Nevertheless, lets see how this goes

### CODE REVIEW

**Review by willstenzel — CHANGES_REQUESTED** [review]
Do we want to possibly record a check in/out event here? It might be hard to analyze the check in/out times if some TA's get cleaned out of the queue without an event. 

@NEUDitao any thoughts on this since you were working the check in/out events previously?

**Comment by NEUDitao** [comment]
> Do we want to possibly record a check in/out event here? It might be hard to analyze the check in/out times if some TA's get cleaned out of the queue without an event. 
> 
> @NEUDitao any thoughts on this since you were working the check in/out events previously?

I think we 100% should. especially with data insights in mind, we'd want to show them in pairs

**Comment by NEUDitao** [comment]
> > Do we want to possibly record a check in/out event here? It might be hard to analyze the check in/out times if some TA's get cleaned out of the queue without an event.
> > @NEUDitao any thoughts on this since you were working the check in/out events previously?
> 
> I think we 100% should. especially with data insights in mind, we'd want to show them in pairs

Though, something else to consider is we could also make a separate type of event for midnight checkouts...

**Comment by willstenzel** [comment]
> > > Do we want to possibly record a check in/out event here? It might be hard to analyze the check in/out times if some TA's get cleaned out of the queue without an event.
> > > @NEUDitao any thoughts on this since you were working the check in/out events previously?
> > 
> > 
> > I think we 100% should. especially with data insights in mind, we'd want to show them in pairs
> 
> Though, something else to consider is we could also make a separate type of event for midnight checkouts...

Good idea. That way we when we go to find out how long TA's are spending in office hours we can compute it correctly by disregarding the check in/out times for the "cleaned TA's".

File: packages/server/src/queue/queue-clean/queue-clean.service.spec.ts [hunk 1]
```diff
@@ -100,6 +101,34 @@ describe('QueueService', () => {
...
+      expect(updatedQueue2.staffList.length).toEqual(0);
+    });
+
+    it('cleaning the queue removes the queue notes', async () => {
+      const ofs = await ClosedOfficeHourFactory.create();
+      const queue = await QueueFactory.create({
+        officeHours: [ofs],
+        notes: 'This note is no longer relevant',
+      });
+    });
```
[the remark this answers is not in the document]
Comment (dfarooq610) [reply packages/server/src/queue/queue-clean/queue-clean.service.spec.ts]: Ahh ok its gone now

**Review by willstenzel — APPROVED** [review]
LGTM

**Comment by liustanley** [comment]
- [ ] migration
- [ ] 3am EST checkout, doesn't checkout if overlaps with scheduled office hour

File: packages/server/src/queue/queue-clean/queue-clean.service.spec.ts [hunk 2]
```diff
@@ -100,6 +104,55 @@ describe('QueueService', () => {
       expect(question.status).toEqual('Stale');
     });
 
+    it.only('checkout all staff from all queues', async () => {
```
Comment (willstenzel) [inline packages/server/src/queue/queue-clean/queue-clean.service.spec.ts:107]: Remove `it.only`

**Review by NEUDitao — APPROVED** [review]

**Pull request merged**